### PR TITLE
Swift: extract missing cases of `AccessorKind` and `AccessSemantics` enums

### PR DESCRIPTION
**Pull request by d10c**
This resolves the "missing switch case" warnings that were showing up while compiling the swift extractor pack.

### CODE REVIEW

**Review by MathiasVP — DISMISSED** [review]
LGTM! I've been meaning to fix this for a long time. Thanks for doing it 🙇!

For future reference: When creating PRs that consist of both autogenerated and manually-written code it's super helpful when the PR is split into commits that are clearly marked as "this commit is totally autogenerated", or "this commit is totally handwritten".

**Comment by d10c** [comment]
Right, I'll get onto the upgrade script. FYI the .cpp and .py files are the manually written ones.

**Comment by AlexDenisov** [comment]
I'm afraid it requires database upgrade/downgrade script 😅

**Comment by d10c** [comment]
Wrote the upgrade script. Though I'm not sure if it really works for updating old DBs, since I get the old "The database may be too new for the QL libraries the query is using; try upgrading them" error when following the testing steps [here](https://github.com/github/codeql/blob/main/docs/prepare-db-upgrade.md) 🤔

**Comment by d10c** [comment]
Updated tests + `Accessor.toString()`. Can confirm that both upgrade and downgrade scripts get applied according to the steps [here](https://github.com/github/codeql/blob/main/docs/prepare-db-upgrade.md?rgh-link-date=2022-11-28T18%3A35%3A48Z).

**Comment by MathiasVP** [comment]
LGTM once the test changes are accepted 😄.

File: swift/extractor/remapping/SwiftOpenInterception.macOS.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ static std::string originalHashFile(const fs::path& filename) {
 }
 
 static int codeql_open(const char* path, int oflag, ...) {
-  va_list ap = {0};
+  va_list ap;
```
Comment (redsun82) [inline swift/extractor/remapping/SwiftOpenInterception.macOS.cpp:30]: why was this changed?

**Review by redsun82 — APPROVED** [review]
LGTM!

**Comment by AlexDenisov** [comment]
I also started an experiment here just to be sure there are no surprises https://github.com/github/codeql-dca-main/issues/8905

File: swift/extractor/remapping/SwiftOpenInterception.macOS.cpp (hunk 1 above)
Comment (d10c) [reply swift/extractor/remapping/SwiftOpenInterception.macOS.cpp:30]: It was showing up as a warning, so I just piggybacked the fix on this PR.

**Comment by MathiasVP** [comment]
DCA looks fine. Merging!
